bankingapp/handler/account/bankaccounthandler.go

```diff
@@ -221,16 +231,49 @@ func Withdraw(ctx *gin.Context) {
 		return
 	}
 
-	// Update the account with new balance
-	_, err = tx.ExecContext(ctx, "UPDATE bankdb.bank_account SET balance = balance - ? WHERE account_id = ?",
-		withdrawAmount, accountId)
-	if err != nil {
-		fail(err)
-		return
-	}
-
 	newBalance := openingBalance - withdrawAmount
 
+	if accType == "FIXED" {
+		//calculate FD maturity date
+		currentDate := time.Now()
+		layout := "2006-01-02 15:04:05"
+		t, err := time.Parse(layout, lockedUntil)
+		if err != nil {
+			fail(errors.New("Error in converting FD Maturity date"))
+			return
+		}
+		//fdMaturityDate := t.Format(layout)
+
+		if currentDate.Before(t) && !preMatureWithdrawal {
+			fail(fmt.Errorf("Locking period of your FD is still not complete."))
+			return
+		} else {
+			if preMatureWithdrawal && withdrawAmount == openingBalance {
+				penalty := openingBalance * 0.1
```

Comment:
penalty should be calculated on the basis of penalty_fd %. dont hardcode it here. 

penalty := openingBalance * penalty_fd/100